### ISSUE TITLE
Message sending failed to UNR addresses

### ISSUE THREAD

**Issue by dimm0** [foxcpp/maddy#287]
# Describe the bug

Getting the error when sending emails to unr.edu domain. The domain works fine in general.

dig for the domain returns the different address, so not sure where _25._tcp is coming from:

```
 ~ dig unr.edu MX

; <<>> DiG 9.10.6 <<>> unr.edu MX
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 7163
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;unr.edu.			IN	MX

;; ANSWER SECTION:
unr.edu.		300	IN	MX	10 unr-edu.mail.protection.outlook.com.
```

# Steps to reproduce

Send an email to unr.edu address

# Log files

```
remote: cannot use MX	{"domain":"unr.edu","msg_id":"b59b1612-5f9b2348","reason":"dns: rcode SERVFAIL when looking up _25._tcp.unr-edu.mail.protection.outlook.com.","remote_server":"unr-edu.mail.protection.outlook.com.","tls_err":null}
queue: delivery attempt failed	{"domain":"unr.edu","msg_id":"b59b1612","rcpt":"earslan@unr.edu","reason":"dns: rcode SERVFAIL when looking up _25._tcp.unr-edu.mail.protection.outlook.com.","smtp_code":451,"smtp_enchcode":"5.4.0","smtp_msg":"No usable MXs, last err: dns: rcode SERVFAIL when looking up _25._tcp.unr-edu.mail.protection.outlook.com.","target":"remote","tls_err":null}
```

# Environment information

* maddy version: 0.4.0

**Comment by foxcpp** [comment]
_25._tcp is coming from the DANE check, unfortunately SERVFAIL error here cannot be ignored as it may be a sign of broken DNSSEC signature and therefore an attempted downgrade attack. 

You can disable DANE, but I suggest you to research the reason SERVFAIL is happening - the issue is most likely outside of maddy.

**Comment by foxcpp** [comment]
Specifically, the following dig command shows the same error when used with 8.8.8.8:
```
dig TLSA _25._tcp.unr-edu.mail.protection.outlook.com.
```

**Comment by foxcpp** [comment]
```
$ dig @ns1-proddns.glbdns.o365filtering.com _25._tcp.unr-edu.mail.protection.outlook.com tlsa

; <<>> DiG 9.16.5 <<>> @ns1-proddns.glbdns.o365filtering.com _25._tcp.unr-edu.mail.protection.outlook.com tlsa
; (2 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: FORMERR, id: 61090
;; flags: qr rd; QUERY: 0, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; WARNING: EDNS query returned status FORMERR - retry with '+noedns'

;; Query time: 90 msec
;; SERVER: 104.47.72.81#53(104.47.72.81)
;; WHEN: Fri Oct 30 06:38:52 MSK 2020
;; MSG SIZE  rcvd: 12

$ dig @ns1-proddns.glbdns.o365filtering.com _25._tcp.unr-edu.mail.protection.outlook.com tlsa +noedns

; <<>> DiG 9.16.5 <<>> @ns1-proddns.glbdns.o365filtering.com _25._tcp.unr-edu.mail.protection.outlook.com tlsa +noedns
; (2 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOTIMP, id: 63714
;; flags: qr rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;_25._tcp.unr-edu.mail.protection.outlook.com. IN TLSA

;; Query time: 100 msec
;; SERVER: 104.47.72.81#53(104.47.72.81)
;; WHEN: Fri Oct 30 06:39:03 MSK 2020
;; MSG SIZE  rcvd: 62
```

Might be an issue on Outlook side.

**Comment by foxcpp** [comment]
Using 1.1.1.1 as a recursor instead of 8.8.8.8 also seems to solve the problem. Perhaps there is a conformance issue on Google DNS side. You did not specify the resolver you are using though. 

Upd: Quad9 public resolver also returns SERVFAIL.

**Comment by dimm0** [comment]
Awesome, thanks! I'll use 1.1.1.1 and will let the admins know
I was indeed using 8.8.8.8 on that server

**Comment by dimm0** [comment]
I'm running maddy in kubernetes cluster. I replaced the DNS with 1.1.1.1 (seems like our local one is also broken), and it's still producing the same error. I just installed dig inside the maddy container, and it's showing the answer with no error. I'm configsed now.. Is maddy using it's own DNS somehow? Or what else can I check?

**Comment by dimm0** [comment]
Fixed by enforcing the 1.1.1.1 DNS in the container. Sorry for the noise.

**Comment by foxcpp** [comment]
The correct fix on maddy side is b6cfad3285b3abafdca5bc4fadc619c588930810.

**Comment by vdukhovni** [comment]
Breakage for all `customer-domain.mail.protection.outlook.com` domains is sadly expected, because Microsofts DNS load-balancer is incorrectly returning `NOTIMP` rather than `NXDOMAIN` for TLSA lookups. However, this is anticipated in RFC7672 which specifies avoiding TLSA lookups for zones in which the A and AAAA RRs of the MX host are unsigned (unless a CNAME into another zone with the initial CNAME signed, requiring a separate lookup to check).
